TicketsDemo.XML/HolidayRepository.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.Serialization;
+using System.IO;
+
+namespace TicketsDemo.XML
+{
+    public class HolidayRepository : IHolidayRepository
+    {
+        XMLSettingsService xml_set = new XMLSettingsService();
```

Comment:
please create interface IXMLSettings, and make XMLSettingsService implementing this service.

Then all over the places XMLSettingsService is used - replace it's usage with usage of IXMLSettings service, injected through DI.